### PR TITLE
DB RPC driver: add master/replica support

### PR DESCRIPTION
**Pull request by agnivade**
We pass a string constant to indicate whether
to get the connection from a replica or a master DB.

It is an untyped string constant to avoid registering
the type or converting from string.

https://focalboard-community.octo.mattermost.com/workspace/zyoahc9uapdn3xdptac6jb69ic?id=285b80a3-257d-41f6-8cf4-ed80ca9d92e5&v=495cdb4d-c13a-4992-8eb9-80cfee2819a4&c=c7386db7-65fd-469b-8bcf-8dc8f8e61e4f

```release-note
NONE
```

### CODE REVIEW

File: shared/driver/driver.go [hunk 1]
```diff
@@ -20,18 +20,24 @@ var (
...
 type Connector struct {
-	api plugin.Driver
+	api    plugin.Driver
+	dbType string
 }
 
-func NewConnector(api plugin.Driver) *Connector {
-	return &Connector{api: api}
+func NewConnector(api plugin.Driver, dbType string) *Connector {
+	return &Connector{api: api, dbType: dbType}
```
Comment (streamer45) [inline shared/driver/driver.go:36]: Purely thinking out loud here: if we know the type is (and will be) a binary decision why don't we explicitly make it so by using a boolean instead?

Comment (agnivade) [reply shared/driver/driver.go:36]: 0/5. I am not sure if it "always will be" a binary decision. Just keeping space for future. But if you prefer to keep it boolean and then change to string later, I can do that.

Comment (streamer45) [reply shared/driver/driver.go:36]: I think I am not entirely convinced by the naming as I implicitly identify the type of a database as something else entirely and I guess changing to boolean would avoid this issue in the first place. But I'll defer to @isacikgoz and see what he thinks as well before we make any change.

Comment (agnivade) [reply shared/driver/driver.go:36]: Completely 0/5 here. Let me know what you guys think is the best.

**Review by streamer45 — APPROVED** [review]
Looks good :+1:

File: shared/driver/driver.go (hunk 1 above)
Comment (isacikgoz) [reply shared/driver/driver.go:36]: 0/5 on either but if I was implementing this I'd pick a boolean for this 😅

**Review by isacikgoz — APPROVED** [review]
Great stuff, this is getting much better every day 🎉

File: shared/driver/driver.go (hunk 1 above)
Comment (agnivade) [reply shared/driver/driver.go:36]: Boolean it is then.

**Pull request merged**